samples/sample_x86_32_gdt_and_seg_regs.c

```diff
@@ -0,0 +1,290 @@
+/*
+
+Sample code to setup a GDT, and use segments.
+
+Copyright(c) 2016 Chris Eagle
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+version 2 as published by the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+*/
+
+#include <unicorn/unicorn.h>
+#include <inttypes.h>
+#include <assert.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+struct SegmentDescriptor {
+   union {
+      struct {   
+# if __BYTE_ORDER == __LITTLE_ENDIAN
```

Comment:
It matters because if I have built unicorn to run on a big endian machine, say an old ppc mac, the fields need to be packed in the descriptor in the proper order for the x86 emulator to pull them back out of memory. I don't have big endian hardware to test on, but if it truly doesn't matter, feel free to trim that part out.